float32[] target_positions
bool wait
---